algorithms.py

```diff
@@ -812,6 +837,21 @@ def common_ancestor_event(self, population_index):
         z = None
         coalescence = False
         defrag_required = False
+        if self.full_arg:
+            self.store_node(population_index)
+            u = len(self.tables.nodes) - 1
+            self.store_edge(x.left, x.right, u, x.node)
```

Comment:
Yeah, this is basically doing the same thing, isn't it? So, we can call ``store_edges_right`` on x and y to do the work here.

Reply:
Hmm, but what happens when we have a coalescence? Are the segments actually getting zipped to together correctly with the new node IDs. I'll have to think about that one a bit.